pkg/provider/terraform/instance/main.go

```diff
@@ -35,11 +36,12 @@ func main() {
 	name := cmd.Flags().String("name", "instance-terraform", "Plugin name to advertise for discovery")
 	logLevel := cmd.Flags().Int("log", cli.DefaultLogLevel, "Logging level. 0 is least verbose. Max is 5")
 	dir := cmd.Flags().String("dir", getDir(), "Dir for storing plan files")
+	pollInterval := cmd.Flags().Duration("poll-interval", 30*time.Second, "Terraform polling interval")
 	cmd.Run = func(c *cobra.Command, args []string) {
 		mustHaveTerraform()
 
 		cli.SetLogLevel(*logLevel)
-		cli.RunPlugin(*name, instance_plugin.PluginServer(NewTerraformInstancePlugin(*dir)))
+		cli.RunPlugin(*name, instance_plugin.PluginServer(NewTerraformInstancePlugin(*dir, *pollInterval)))
```

Comment:
Instead adding the `pollInterval` param to the function `NewTerraformInstancePlugin`, you could just add the function here that is required for plugin lookup:  https://github.com/docker/infrakit/blob/master/cmd/infrakit/main.go#L97

Also be sure to add a few lines of code to initialize the plugin discovery like so: https://github.com/docker/infrakit/blob/master/cmd/infrakit/main.go#L63

Note here there's no need to worry about "remote" plugin look up in this case.  Just call the local plugin lookup like so: https://github.com/docker/infrakit/blob/master/cmd/infrakit/main.go#L106

Once you have the `func() plugin.Plugins` plumbed into the Terraform plugin, you can pass that to the manager discovery function `pkg/manager/discovery/Locate()` to get a handle on the local manager running on the same node.  Once you have that you can call `IsLeader()` and run / not run `doTerraformApply` as appropriate.